### DISCUSSION TITLE
Offline persistence

### DISCUSSION THREAD

**Discussion by rjsandman** [urql-graphql/urql#1807] [Q&A]
When setting up an offlineExchange, I can see the indexDB loading with the correct data.  Using chrome dev tools if I turn the browser offline and refresh I was expecting the page to load data from indexDB but the indexDB gets wiped clean.  Is this the expected result or should the indexDB persist on a page reload when offline?  

Thanks!

**Comment by JoviDeCroock** [comment]
IndexedDB persists across sessions by default, I haven't encountered an issue with us clearing that yet 😅

**Reply by rjsandman** [reply]
Hmm... Is there a reason that the indexDB would get cleared on page refresh?

**Reply by JoviDeCroock** [reply]
No, maybe some setting you have?

So I tried it in this sandbox: https://codesandbox.io/s/zen-albattani-ic59e?file=/src/index.js:550-563 I loaded this page in the url provided on the right initially, went back in the code and removed the `fetchExchange` and it all still worked.

**Reply by rjsandman** [reply]
Ahh. Okay, that works for me as well.  I was under the impression that a page refresh would still persist the indexDB but it appears that it will not.

**Reply by JoviDeCroock** [reply]
It does persist the indexedDB for me 😅 but maybe it's never started due to your JS bundle not being available during offline? Just hypothesising as it's hard without a means of reproducing this.

**Reply by rjsandman** [reply]
Got it!  Thanks!
